src/utils/calendar_manager.ts

```diff
@@ -376,6 +380,215 @@ const buildMeetingData = async (
+    if (isValidEmail(meetingUrl)) {
+      throw new InvalidURL()
+    }
+    if (!isValidUrl(meetingUrl)) {
+      meetingUrl = `https://${meetingUrl}`
+      if (!isValidUrl(meetingUrl)) {
+        throw new InvalidURL()
+      }
+    }
+  }
+
+  const privateInfo: MeetingInfo = {
+    created_at: new Date(),
+    participants: sanitizedParticipants,
+    title: meetingTitle,
+    content: meetingContent,
+    meeting_url: meetingUrl,
+    change_history_paths: [],
+    related_slot_ids: [],
+    meeting_id: meetingId,
+    reminders: meetingReminders,
+    provider: meetingProvider,
+    recurrence: meetingRepeat,
+    permissions: selectedPermissions,
+  }
+  // first pass to make sure that we are keeping the existing slot id
+  for (const participant of sanitizedParticipants) {
+    const existingSlotId =
+      participantsToKeep?.[
+        participant.account_address || participant.guest_email || ''
+      ]
+    participant.slot_id = existingSlotId || participant.slot_id
+  }
+
+  const allSlotIds = sanitizedParticipants
+    .map(it => it.slot_id!)
+    .filter(val => val !== undefined)
+
+  const participantsMappings = []
+
+  for (const participant of sanitizedParticipants) {
+    const encodingKey = process.env.NEXT_PUBLIC_SERVER_PUB_KEY!
+
+    const privateInfoComplete = JSON.stringify({
+      ...privateInfo,
+      related_slot_ids: allSlotIds.filter(id => id !== participant.slot_id),
+    } as MeetingInfo)
+
+    const participantMapping: RequestParticipantMapping = {
+      account_address: participant.account_address || '',
+      slot_id: participant.slot_id,
+      type: participant.type,
+      meeting_id: meetingId,
+      privateInfo: await encryptWithPublicKey(encodingKey, privateInfoComplete),
+      privateInfoHash: simpleHash(privateInfoComplete),
+      timeZone: Intl.DateTimeFormat().resolvedOptions().timeZone,
+      name: participant.name || '',
+      guest_email: participant.guest_email,
+      status:
+        participant.type === ParticipantType.Scheduler
+          ? ParticipationStatus.Accepted
+          : ParticipationStatus.Pending,
+      mappingType: !!participantsToKeep?.[
+        participant.account_address || participant.guest_email || ''
+      ]
+        ? ParticipantMappingType.KEEP
+        : ParticipantMappingType.ADD,
+    }
+
+    participantsMappings.push(participantMapping)
+  }
+
+  return {
+    type: schedulingType,
+    start: startTime,
+    end: endTime,
+    meeting_id: meetingId,
+    participants_mapping: participantsMappings,
+    meetingTypeId: '',
+    meeting_url: privateInfo['meeting_url'],
+    content: privateInfo['content'],
+    title: privateInfo['title'],
+    meetingProvider,
+    meetingReminders,
+    meetingRepeat,
+    allSlotIds,
+    meetingPermissions: selectedPermissions,
+    ignoreOwnerAvailability: false,
+  }
+}
+
+const updateMeetingAsGuest = async (
+  slotId: string,
+  startTime: Date,
+  endTime: Date,
+  participants: ParticipantInfo[],
```

Comment:
The participants passed from updating should only be new participants say the user added a new guest email to the meeting. The rest of the guests should be gotten from mapping the acting Slots